gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/DefaultBigQueryRpc.java

```diff
@@ -0,0 +1,421 @@
+  }
+
+  @Override
+  public Dataset getDataset(String datasetId, Map<Option, ?> options) throws BigQueryException {
+    try {
+      return bigquery.datasets()
+          .get(this.options.projectId(), datasetId)
+          .setFields(FIELDS.getString(options))
+          .setQuotaUser(QUOTA_USER.getString(options))
+          .setUserIp(USER_IP.getString(options))
+          .execute();
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<Dataset>> listDatasets(Map<Option, ?> options)
+      throws BigQueryException {
+    try {
+      DatasetList datasetsList = bigquery.datasets()
+          .list(this.options.projectId())
+          .setQuotaUser(QUOTA_USER.getString(options))
+          .setUserIp(USER_IP.getString(options))
+          .setAll(Option.ALL_DATASETS.getBoolean(options))
+          .setMaxResults(MAX_RESULTS.getLong(options))
+          .setPageToken(PAGE_TOKEN.getString(options))
+          .execute();
+      Iterable<DatasetList.Datasets> datasets = datasetsList.getDatasets();
+      return Tuple.of(datasetsList.getNextPageToken(),
+          Iterables.transform(datasets != null ? datasets : ImmutableList.<DatasetList.Datasets>of(),
+              new Function<DatasetList.Datasets, Dataset>() {
+                @Override
+                public Dataset apply(DatasetList.Datasets f) {
+                  return new Dataset()
+                      .setDatasetReference(f.getDatasetReference())
+                      .setFriendlyName(f.getFriendlyName())
+                      .setId(f.getId())
+                      .setKind(f.getKind());
+                }
+              }));
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Dataset create(Dataset dataset, Map<Option, ?> options) throws BigQueryException {
+    try {
+      return bigquery.datasets().insert(this.options.projectId(), dataset)
+          .setFields(FIELDS.getString(options))
+          .setQuotaUser(QUOTA_USER.getString(options))
+          .setUserIp(USER_IP.getString(options))
+          .execute();
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public boolean deleteDataset(String datasetId, Map<Option, ?> options) throws BigQueryException {
+    try {
+      bigquery.datasets().delete(this.options.projectId(), datasetId)
+          .setQuotaUser(QUOTA_USER.getString(options))
+          .setUserIp(USER_IP.getString(options))
+          .setDeleteContents(DELETE_CONTENTS.getBoolean(options))
+          .execute();
+      return true;
+    } catch (IOException ex) {
+      BigQueryException serviceException = translate(ex);
+      if (serviceException.code() == 404) {
+        return false;
+      }
+      throw serviceException;
+    }
+  }
+
+  @Override
+  public Dataset patch(Dataset dataset, Map<Option, ?> options) throws BigQueryException {
```

Comment:
I thought that issue only applied to `XXXService` classes and that in `XXXRpc` we should be keeping the name of the rpc. I might have got it wrong though.